core/keyboardshortcut_registry.js

```diff
@@ -0,0 +1,351 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview The namespace used to keep track of keyboard shortcuts and the
+ * key codes used to execute those shortcuts.
+ * @author aschmiedt@google.com (Abby Schmiedt)
+ */
+'use strict';
+
+goog.provide('Blockly.KeyboardShortcutRegistry');
+
+goog.require('Blockly.KeyboardShortcutItems');
+goog.require('Blockly.navigation');
+goog.require('Blockly.utils.object');
+
+
+/**
+ * Class for the registry of keyboard shortcuts. This is intended to be a
+ * singleton. You should not create a new instance, and only access this class
+ * from Blockly.KeyboardShortcutRegistry.registry.
+ * @constructor
+ */
+Blockly.KeyboardShortcutRegistry = function() {
+  // Singleton instance should be registered once.
+  Blockly.KeyboardShortcutRegistry.registry = this;
+
+  /**
+   * Registry of all keyboard shortcuts, keyed by name of shortcut.
+   * @type {!Object<string, !Blockly.KeyboardShortcutRegistry.KeyboardShortcut>}
+   * @private
+   */
+  this.registry_ = {};
```

Comment:
Safest to use Object.create(null) here and we won't have to worry about inherited properties.

Reply:
done